quickwit/quickwit-control-plane/src/control_plane.rs

```diff
@@ -772,6 +778,49 @@ impl Handler<DeleteSourceRequest> for ControlPlane {
     }
 }
 
+#[async_trait]
+impl Handler<DebouncedPruneShardsRequest> for ControlPlane {
+    type Reply = ControlPlaneResult<EmptyResponse>;
+
+    async fn handle(
+        &mut self,
+        request: DebouncedPruneShardsRequest,
+        ctx: &ActorContext<Self>,
+    ) -> Result<ControlPlaneResult<EmptyResponse>, ActorExitStatus> {
+        let metastore_request = request.request.unwrap();
+        if request.execute_now {
+            self.metastore
+                .prune_shards(metastore_request)
+                .await
+                .context("failed to prune shards")?;
+            return Ok(Ok(EmptyResponse {}));
+        }
+
+        let debouncer = self
+            .prune_shards_debouncers
+            .entry((
+                metastore_request.index_uid().index_id.clone(),
+                metastore_request.source_id.clone(),
+            ))
+            .or_insert_with(|| Debouncer::new(PRUNE_SHARDS_DEFAULT_COOLDOWN_PERIOD));
+
+        // Update the cooldown period in case the config changes. If multiple
+        // indexers run with different configs, the control plane ends up
+        // running one of them in a seemingly arbitrary way.
+        if let Some(interval) = metastore_request.interval {
+            debouncer.set_next_cooldown_period(Duration::from_secs(interval as u64));
+        }
+        debouncer.self_send_with_cooldown(
+            ctx,
+            DebouncedPruneShardsRequest {
```

Comment:
Shouldn't this be two messages? DebouncedPruneShardRequests and PruneShardsRequest ?
(without a execute_now boolean)

Reply:
I hesitated. 2 messages are probably a bit more understandable.